query-graphs-js/src/pathTree.ts

```diff
@@ -321,7 +333,7 @@ export class PathTree<TTrigger, RV extends Vertex = Vertex, TNullEdge extends nu
     }
     assert(addIdx === newSize, () => `Expected ${newSize} childs but only got ${addIdx}`);
 
-    return new PathTree(this.graph, this.vertex, localSelections, this.triggerEquality, newChilds);
+    return new PathTree(this.graph, this.vertex, localSelections, this.triggerEquality, newChilds, mergedContextToSelection, mergedParameterToContext);
```

Comment:
Note that in addition to updating `merge()`, you'll also need to update `equalsSameRoot()` (a method whose job is to basically do a fast check to see whether two `PathTree`s are equal).